### ISSUE TITLE
ZSH Completion Script incorrect

### ISSUE THREAD

**Issue by MarcoEidinger** [apple/swift-package-manager#5789]
### Description

Trying to use auto-completion for `swift package` (or other SwiftPM commands) relying on completion script generated with  `swift package completion-tool generate-zsh-script`

### Expected behavior

Code completion appears

### Actual behavior

Getting parsing error when trying code completion

![Code Completion Error](https://user-images.githubusercontent.com/4176826/183760243-fc3b9b7e-c8e4-4dc5-8bdb-9f5a0e18015e.gif)

### Steps to reproduce

Run the commands as noted in https://github.com/apple/swift-package-manager/blob/main/Documentation/Usage.md#using-shell-completion-scripts

```
mkdir ~/.zsh
swift package completion-tool generate-zsh-script > ~/.zsh/_swift
echo -e "fpath=(~/.zsh \$fpath)\n" >> ~/.zshrc
compinit
```

Then open new terminal and try auto-complete (`TAB`) for `swift package`

### Swift Package Manager version/commit hash

Swift 5.6.1 (Xcode 13.4.1)

### Swift & OS version (output of `swift --version && uname -a`)

swift-driver version: 1.45.2 Apple Swift version 5.6.1 (swiftlang-5.6.0.323.66 clang-1316.0.20.12)
Target: arm64-apple-macosx12.0

**Comment by tomerd** [comment]
@bitjammer is this something you are familiar with?

**Comment by neonichu** [comment]
We're just calling `ParsableArguments.completionScript(for:)` from argument parser for this, btw.

**Comment by rauhul** [comment]
If you could reduce this to a small reproducer and open an issue in apple/swift-argument-parser we can take a look!

**Comment by MarcoEidinger** [comment]
It appears that using an empty string as a value for `commandName` in `DefaultCommand` causes the issue.

https://github.com/apple/swift-package-manager/blob/6aff18f6e2f7372cd879e0c2829e1c75378c75ee/Sources/Commands/SwiftPackageTool.swift#L1480

In my small reproducer (https://github.com/MarcoEidinger/spm-issue-5789-repo) I find no issues when using ` commandName: nil,`

@rauhul please advise if you see this as issue for apple/swift-argument-parser and I should create an issue there OR if you think that SPM shall change their implementation

FYI @neonichu

**Comment by rauhul** [comment]
Looks like a sap bug imo, though the quick solution for spm is changing the default to nil. I can probably fix the sap issue tomorrow morning, but filing an issue would be good regardless! 

Thanks a ton for tracking down the bug 💛

**Comment by MarcoEidinger** [comment]
I created PR to change SPM for an easy solution.

SPM is using swift-argument-parser@1.0.3

@rauhul further testing reveals that this particular issue is solved in swift-argument-parser@1.1.4 due to https://github.com/apple/swift-argument-parser/pull/443 which filters out subcommands that shall not be displayed which would be the case for SPM's `DefaultCommand`

```swift
extension SwiftPackageTool {
    // This command is the default when no other subcommand is passed. It is not shown in the help and is never invoked directly.
    struct DefaultCommand: SwiftCommand {
        static let configuration = CommandConfiguration(
            commandName: "",
            shouldDisplay: false) // <====!!!!!!!!!!!
         }
     }
}
```

Hence I won't file an issue in swift-argument-parser

**Comment by rauhul** [comment]
I think this is still reveals a bug in sap as we should never generate illegal completion scripts.
